### PR TITLE
feat(DDI-1282): callout updates with medium size and replace spacing …

### PR DESCRIPTION
**Pull request by vanessatran-ddi**
### Description

- Update the callout component so that it is using our spacing tokens
- Include small callout variant at “mobile” breakpoint (using current mobile breakpoint)
- Small callout will be available as a variant that can be specified on any viewport size

Here is the [latest figma ](https://www.figma.com/file/EFk8pSKbRtTbUvS2hWldSY/Component---Small-Callouts?t=sgwOuTaM35UcmAC0-0) (Confirmed with Thomas) 

image

### Story 
[DDI-1282](https://goa-dio.atlassian.net/browse/DDIDS-1282)

### Demo link: 
[https://vanessatran-ddi.github.io/angular-ui-component-sample/home](https://vanessatran-ddi.github.io/angular-ui-component-sample/home)
#### Angular app
http://localhost:4200/callout
image

#### React app
http://localhost:4200/callout
image

#### Story book
http://localhost:4400/?path=/docs/components-callout--emergency

image
image

#### Breakpoint (Mobile)

The large callout will be updated to "medium" style if the screen size is < 624px.
(Requirement as below)

image

Demo Video is as below: 

https://user-images.githubusercontent.com/120135417/231512920-6db0ccf0-1796-44ad-8993-c1aef733e1c3.mov

#### What are changed
- Add a new attribute size for the Callout. We have 2 sizes: large, and medium. Large is a default value. 
- Add a breakpoint automatically to make the "Large" callout becomes a "Small" callout. 
- Storybook is updated. Angular and React examples are updated. 
- Unit tests are added.

### CODE REVIEW

**Comment by Radha-Alberta** [comment]
- [ ] For medium callout, there should be 4px spacing  to the left and right of the icon. Currently, both medium and large callouts has same spacing. (Image 1)

#### Image 1
image

**Comment by vanessatran-ddi** [comment]
> * [x]  For medium callout, there should be 4px spacing  to the left and right of the icon. Currently, both medium and large callouts has same spacing. (Image 1)
> * [x]  ~all of the corners currently have a radius, in callouts all borders should be square~
> * [x]  Design tokens should be --goa-typography-body-s for the body text and --goa-typography-heading-xs for the heading
> * [x]  ~Should be 4px (--goa-spacing-2xs) between the heading and the body text~
> * [x]  Icon should be aligned with the heading/body of medium and large callouts. (Image 2)
> 
> #### Image 1
> image
> 
> #### Image 2
> image

image

File: libs/web-components/package-lock.json
Comment (ArakTaiRoth) [inline libs/web-components/package-lock.json:1]: I think you need to undo the changes to the `package-lock` and the `package.json` files, as there's no change to the `package.json` that should have a corresponding change to the `package-lock`.

File: libs/web-components/src/components/callout/Callout.svelte [hunk 1]
```diff
@@ -71,15 +87,16 @@
   .notification {
     display: flex;
     align-items: stretch;
-    border-radius: 3px;
```
Comment (ArakTaiRoth) [inline libs/web-components/src/components/callout/Callout.svelte:74]: Why the removal of `border-radius`?

Comment (vanessatran-ddi) [reply libs/web-components/src/components/callout/Callout.svelte:74]: @ArakTaiRoth It is an updated UI version from Thomas. 
image

File: libs/web-components/src/components/callout/Callout.svelte [hunk 2]
```diff
@@ -13,50 +13,66 @@
...
       ? "warning"
       : type === "important"
-      ? "alert-circle"
-      : type === "information"
-      ? "information-circle"
-      : type === "success"
-      ? "checkmark-circle"
-      : type === "event"
-      ? "calendar"
-      : "";
```
Comment (chrisolsen) [inline libs/web-components/src/components/callout/Callout.svelte:41]: Can you revert this spacing. Having things with the new tabbing makes things harder to read.

**Review by ArakTaiRoth — DISMISSED** [review]

File: libs/web-components/src/components/callout/Callout.svelte [hunk 3]
```diff
@@ -13,19 +13,27 @@
...
   export let ml: Spacing = null;
-
+  export let size: CalloutSize = "large";
   export let type: CalloutType;
   export let heading: string = "";
   export let testid: string = "";
+  let screenSize = 0;
+  let iconSize = "medium";
+
+  $: isMediumCallout = screenSize < 624 || size === "medium";
```
[the remark this answers is not in the document]
Comment (vanessatran-ddi) [reply libs/web-components/src/components/callout/Callout.svelte:36]: Yes it is defined under figma from Tom. 
image
It is intentional based on the documentation.

File: libs/web-components/src/components/callout/Callout.spec.ts [hunk 4]
```diff
@@ -82,6 +82,18 @@ describe('GoACalloutComponent', () => {
...
+    const el = baseElement.queryByTestId("testid");
+
+    const heading = el.querySelector('h3.medium');
+    expect(heading).toHaveClass("medium");
+
+    setTimeout(() => {
+      const icon = el.querySelector('goa-icon');
+      expect(icon).toHaveAttribute("size", "small");
+    });
+  });
```
Comment (chrisolsen) [inline libs/web-components/src/components/callout/Callout.spec.ts:96]: Ideally all the quotes should be double.

File: libs/web-components/src/components/callout/Callout.svelte (hunk 3 above)
Comment (chrisolsen) [inline libs/web-components/src/components/callout/Callout.svelte:36]: Is there a reason the screen size is being obtained in this manner instead of using a media query?

Comment (chrisolsen) [reply libs/web-components/src/components/callout/Callout.svelte]: That makes sense. We might see an issue with this when `@container` queries are added, which was just added in the most recent version of svelte, but we can deal with that later.

**Review by chrisolsen — APPROVED** [review]
